test/test_main.py

```diff
@@ -781,7 +852,8 @@ def _setup_entry_point(self, directory, entry_point):
             path.chmod(0o755)
 
     def _call_event(self, rel_path, env):
-        env["JUJU_DISPATCH_PATH"] = str(rel_path)
+        env['JUJU_DISPATCH_PATH'] = str(rel_path)
+        env['JUJU_VERSION'] = '2.8.0'
```

Comment:
I feel like we are missing the test that does the charmcraft mechanism of using an old juju but forcing JUJU_DISPATCH_PATH.
(I don't see a change which is doing:
```
env['JUJU_DISPATCH_PATH'] = ...
env['JUJU_VERSION'] = '2.7.0'
```

Reply:
I missed that you set the default JUJU_VERSION to 2.7.0, so maybe it is already covered.

Reply:
you're right, we're not covering that one. I'll add it :)